### PR TITLE
clearStages and clearTags buttons, see #10

### PR DESCRIPTION
**Pull request by HaraldAstok**

### CODE REVIEW

File: src/containers/table/Table.js [hunk 1]
```diff
@@ -187,6 +187,30 @@ class TableView extends React.Component {
...
+      unSelectedTags: unSelected,
+      selectedTags: []
+    });
+  }
+
+  clearStages() {
+    const selected = this.state.selectedStages;
+    let unSelected = this.state.unSelectedStages;
+    for (let i = 0; i < selected.length; i++) {
+      unSelected.push(selected[i]);
```
Comment (urbanikm) [inline src/containers/table/Table.js]: same here

File: src/containers/table/Table.js [hunk 2]
```diff
@@ -187,6 +187,30 @@ class TableView extends React.Component {
     });
   }
 
+  clearTags() {
+    const selected = this.state.selectedTags;
+    let unSelected = this.state.unSelectedTags;
+    for (let i = 0; i < selected.length; i++) {
+      unSelected.push(selected[i]);
```
Comment (urbanikm) [inline src/containers/table/Table.js]: After using 'clear tags' tags won't be sorted alphabetically, sort them after you push elements back to unselected tags

Comment (HaraldAstok) [reply src/containers/table/Table.js]: Should be fixed

File: src/containers/table/Table.js (hunk 1 above)
Comment (HaraldAstok) [reply src/containers/table/Table.js]: Should be fixed

File: src/containers/table/Table.js [hunk 3]
```diff
@@ -187,6 +189,36 @@ class TableView extends React.Component {
...
+    }
+    unSelected.sort((a, b) => {
+      return a.toLowerCase().localeCompare(b.toLowerCase());
+    });
+    this.setState({
+      unSelectedStages: unSelected,
+      selectedStages: []
+    });
+  }
+
```
Comment (germohn) [inline src/containers/table/Table.js]: use better option of defining extra variable with initial state, so you could use it here.

File: src/containers/table/Table.js [hunk 4]
```diff
@@ -79,7 +79,9 @@ class TableView extends React.Component {
         return a.toLowerCase().localeCompare(b.toLowerCase());
       }),
       selectedTags: [],
-      unSelectedStages: props.stages,
+      unSelectedStages: props.stages.sort((a, b) => {
```
Comment (germohn) [inline src/containers/table/Table.js]: no point of sorting stages in alphabetical order, they should have their own order

File: src/containers/table/Table.js [hunk 5]
```diff
@@ -205,7 +237,9 @@ class TableView extends React.Component {
     selected.splice(index, 1);
     let unSelected = this.state.unSelectedStages;
     unSelected.push(stage);
-
+    unSelected.sort((a, b) => {
```
Comment (germohn) [inline src/containers/table/Table.js]: same comment about stages

**Review by germohn — CHANGES_REQUESTED** [review]

File: src/containers/table/Table.js [hunk 6]
```diff
@@ -94,9 +94,13 @@ class TableView extends React.Component {
       selectedStages: [],
       companies: R.sortBy(R.compose(R.toLower, R.prop('name')))(props.data),
       search: '',
+
+      alltags: props.tags,
       allstages: Array.from((props.stages).values())
     };
 
+    const temp = props.tags;
```
Comment (urbanikm) [inline src/containers/table/Table.js:102]: isn't this variable the same as alltags? Are both needed?

File: src/containers/table/Table.js [hunk 7]
```diff
@@ -105,6 +109,13 @@ class TableView extends React.Component {
     this.showAll = this.showAll.bind(this);
   }
 
+  onClearState(e) {
```
Comment (urbanikm) [inline src/containers/table/Table.js]: the name of this method is a bit confusing, maybe onClearTags orsmth would be more informative?

**Review by germohn — APPROVED** [review]

**Pull request merged**